.github/workflows/lint-codeowners.yaml

```diff
@@ -1,129 +0,0 @@
-name: CODEOWNERS Checks
```

Comment:
Why aren't we keeping this one?

Reply:
Last time we removed it, so I removed it again this time. I didn't put much thought into it.

Reply:
I'll bring it back. If it causes problems, we can rip it back out.

Reply:
I'm confused, the workflow complains about I think every file, despite there being a catchall `* @cilium/tophat` match in codeowners?

https://github.com/cilium/cilium/actions/runs/7202801453/job/19621632353?pr=29838

Reply:
Yeah it looks like the workflow deliberately ignores the wildcard, but that's always how we've specified codeowners for the stable branches.

The other two jobs in this workflow lint are for checking if there are files referenced in codeowners that don't exist (unlikely to occur on a stale branch, and low stakes) and ensuring the codeowners that are in the file are documented (we don't add new codeowners into stable branches, only remove codeowners to simplify backport review). So I think the right answer is to delete this workflow.